src/plugins/intel_cpu/CMakeLists.txt

```diff
@@ -42,6 +42,10 @@ if(DNNL_USE_ACL)
     set(OV_CPU_WITH_ACL ON)
 endif()
 
+if (OV_CPU_WITH_ACL AND APPLE)
```

Comment:
@EgorDuplensky  @alvoron   In fact condition which arch is used for ACL is determined here: https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/thirdparty/ACLConfig.cmake#L107. So once it will be changed this condition will become invalid. 
Does it make sense to introduce intermidiate fp16 support variable inside ACLConfig which will contribute to OV_CPU_WITH_ACL_FP16 condition there?

Reply:
maybe we need to detect it directly in cpp code? E.g. when compiler enables FP16, it provides some macro like `__ARM_FEATURE_FP16_VECTOR_ARITHMETIC`

Reply:
The intention is the following: the decision on FP16 support on product level (assuming CPU arch) should be made based on the plugin feedback. The plugin itself defines (based os some internal considirations) corresponding variable  and bypass it all other parts of the product (including CPU third-parties).
We don't want to use HW specific flags, as long as HW level FP16 support is not equal to product level FP16 support. Such differentiation might be also usefull for HW-agnostic conditional compilation (e.g. to enable compilation w/o FP16 support).